qcodes/utils/installation_info.py

```diff
@@ -0,0 +1,119 @@
+"""
+This module contains helper functions that provide information about how
+QCoDeS is installed and about what other packages are installed along with
+QCoDeS
+"""
+from typing import Dict, List
+import subprocess
+from pip._vendor import pkg_resources
+import importlib
+
+import qcodes
+
+
+# sometimes a package is imported as something else than its package name
+# _IMPORT_NAMES maps package name to import name
+_IMPORT_NAMES = {'pyzmq': 'zmq'}
+_PACKAGE_NAMES = {v: k for k, v in _IMPORT_NAMES.items()}
+
+# sometimes we import non-versioned packages backported from the standard
+# library (e.g. dataclasses for python 3.6). Those should be excluded from
+# any version listing
+_BACKPORTED_PACKAGES = ['dataclasses']
+
+
+def _pip_list_parser(pip_list_raw: List[str]) -> Dict[str, Dict[str, str]]:
+    """
+    Parse the raw output of pip list into a dict with package name keys and
+    a dict {'version': ..., 'location': ...} values
+
+    Args:
+        pip_list_raw: a list of strings, one line per list element
+    """
+    out: Dict[str, Dict[str, str]] = {}
+
```

Comment:
Pips output format changed some years back. I suspect this may fail for older versions of pip . Do we care.

Reply:
I just discovered that there is also `pip list -e --format json` that could make parsing simpler?

Reply:
Specifically the default changed here https://pip-python3.readthedocs.io/en/stable/reference/pip_list.html#cmdoption-format from legacy to columns at some point. 

Specifying the format upfront is probably a good idea

Reply:
The changes are documented here https://pip.pypa.io/en/stable/news/ 

It seems like columns was added in 9.0 (2016), the default changed in 10.0 (2018) and the legacy format removed in 18.0 (2018).